prow/cmd/build/main.go

```diff
@@ -70,6 +71,7 @@ func (o *options) parse(flags *flag.FlagSet, args []string) error {
 	flags.StringVar(&o.buildCluster, "build-cluster", "", "Path to file containing a YAML-marshalled kube.Cluster object. If empty, uses the local cluster.")
 	flags.StringVar(&o.cert, "tls-cert-file", "", "Path to x509 certificate for HTTPS")
 	flags.StringVar(&o.privateKey, "tls-private-key-file", "", "Path to matching x509 private key.")
+	flags.DurationVar(&o.buildTimeout, "build-timeout", 1*time.Hour, "Default timeout for build. Only used if not timeout is defined in the prow job")
```

Comment:
On Pod jobs we have timeouts per-job, configurable default in Plank config and an ability to override on a per-job level. That may be a better approach here?

Reply:
hummmmm the default podutils timeout should apply to the build jobs as well? no?

Reply:
The issue is when the timeout in the prow job is set to 0 (default). The same value is also set into the build resulting in an infinite timeout. Also the same situation is when the job decoration is not set.

This ensure basically that always there is a timeout, but always the timeout from the job or build has precedence over this if is configured.

Reply:
Why is that a bad timeout value? Is it not just as harmful on Pods as it is on Builds? If a user wants a global default, why wouldn't they set one?

Reply:
The user does not have always control over this value. The prow jobs are created by `tide` with no timeout set. See https://github.com/kubernetes/test-infra/blob/c568296819c5e1b355d8e574284dada33b5c8241/prow/config/config.go#L1198 and https://github.com/kubernetes/test-infra/blob/c568296819c5e1b355d8e574284dada33b5c8241/prow/pjutil/pjutil.go#L108

Reply:
`tide` is just using the configuration that was created by users? In the `PresubmitSpec` function you link to, the `p config.Presubmit` is the configuration on disk for the job, which is written by users.

Reply:
That's right, but imagine that you have 50+ jobs defined in the configuration but some of them are missing the timeout. The intend of this timeout is to increase the reliability of the build. We had this situation in Jenkins X.

Reply:
You can set the default timeout in the `plank` configuration, though, for all jobs. Then only jobs that explicitly override will get anything else.

Reply:
Which timeout are you referring to? `PodPendingTimeout`? As far as I can see in the code, this timeout is only used for pods created by plank for prow jobs with `kubernetes` agent, but not `kanative-build` agent.  

This default timeout is only for `knative-build` controller. I can also add the vaulue in the knative build controller config.

Reply:
> On Pod jobs we have timeouts per-job, configurable default in Plank config and an ability to override on a per-job level. That may be a better approach here?

Yes, please reuse these values.

> PodPendingTimeout? As far as I can see in the code, this timeout is only used for pods created by plank for prow jobs with kubernetes agent, but not kanative-build agent.

Please teach knative-build to use this value, just like we taught it to use PodNamespace. Let's not create a separate value.

Reply:
Yes, when you move it we need to continue loading the old value

Reply:
I removed the flag and define a config value.